### PR TITLE
fix the metadata leak when the Replicated*MergeTree with custom ZooKeeper cluster is dropped

### PR DESCRIPTION
**Pull request by fastio**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix the metadata leak when the Replicated*MergeTree with custom (non default) ZooKeeper cluster is dropped.

Detailed description / Documentation draft:

Fix the bug that the metadata of the replicated family merge tree table were not removed from the auxiliary ZooKeeper when the table dropped.

#21054

### CODE REVIEW

**Comment by fastio** [comment]
I'll add the test case soon.

File: src/Storages/StorageReplicatedMergeTree.cpp [hunk 1]
```diff
@@ -750,7 +750,7 @@ void StorageReplicatedMergeTree::drop()
     {
         /// Table can be shut down, restarting thread is not active
         /// and calling StorageReplicatedMergeTree::getZooKeeper() won't suffice.
-        auto zookeeper = global_context.getZooKeeper();
+        auto zookeeper = getZooKeeper();
```
Comment (cyberhuman) [inline src/Storages/StorageReplicatedMergeTree.cpp:753]: The comment above says "and calling StorageReplicatedMergeTree::getZooKeeper() won't suffice", does it not apply anymore?

**Comment by nvartolomei** [comment]
You can pick tests from here if you want https://github.com/ClickHouse/ClickHouse/pull/20141/files

File: src/Storages/StorageReplicatedMergeTree.cpp (hunk 1 above)
Comment (fastio) [reply src/Storages/StorageReplicatedMergeTree.cpp:753]: @cyberhuman, Thanks. The comment is still correct now. I'll fix it.

**Comment by fastio** [comment]
> You can pick tests from here if you want https://github.com/ClickHouse/ClickHouse/pull/20141/files

@nvartolomei Thanks.

File: tests/integration/test_replicated_merge_tree_with_auxiliary_zookeepers/test.py [hunk 2]
```diff
@@ -78,3 +77,27 @@ def test_create_replicated_merge_tree_with_not_exists_auxiliary_zookeeper(starte
...
+    # Insert data into node1, and query it from node2.
+    node1.query("INSERT INTO test_auxiliary_zookeeper VALUES (1)")
+    time.sleep(5)
+
+    expected = "1\n"
+    assert TSV(node1.query("SELECT a FROM test_auxiliary_zookeeper")) == TSV(expected)
+    assert TSV(node2.query("SELECT a FROM test_auxiliary_zookeeper")) == TSV(expected)
+
+    zk = cluster.get_kazoo_client('zoo1')
+    assert zk.exists('/clickhouse/tables/test/test_auxiliary_zookeeper')
```
Comment (nvartolomei) [inline tests/integration/test_replicated_merge_tree_with_auxiliary_zookeepers/test.py:101]: I think this tests passes even without the fix.

Comment (fastio) [reply tests/integration/test_replicated_merge_tree_with_auxiliary_zookeepers/test.py:101]: @nvartolomei The metadata path should be exists if the table is alive. When the table dropped, the metadata path should be removed from zookeeper.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**